packages/asset-service/src/service/AssetService.ts

```diff
@@ -0,0 +1,60 @@
+import { BaseAsset, Asset, NetworkTypes } from '../types'
+import axios from 'axios'
+import localAssetData from './generatedAssetData.json'
+
+export class AssetService {
+  private assetFileUrl: string
+
+  private assetData: BaseAsset[]
+  private flatAssetData: Asset[]
+
+  constructor(assetFileUrl: string) {
+    this.assetFileUrl = assetFileUrl
+  }
+
+  private checkInitialized() {
+    if (!this.assetData || !this.flatAssetData) throw new Error('Asset service not initialized')
+  }
+
+  /**
+   * Get asset data from assetFileUrl and flatten it for easier use
+   */
+  async initialize() {
+    try {
+      const { data } = await axios.get<BaseAsset[]>(this.assetFileUrl)
+      this.assetData = data
+    } catch (err) {
+      this.assetData = localAssetData as BaseAsset[]
+    }
+
+    const flatAssetData: Asset[] = []
+
+    for (const baseAsset of this.assetData) {
+      const newAsset = { ...baseAsset }
+      delete newAsset.tokens
+      flatAssetData.push(newAsset)
+      if (baseAsset.tokens) {
+        for (const tokenAsset of baseAsset.tokens) {
+          flatAssetData.push({
+            ...tokenAsset,
+            chain: baseAsset.chain,
+            network: baseAsset.network
+          })
+        }
+      }
+    }
```

Comment:
This looks a lot like a `.flatMap`.

Reply:
As far as I can tell flatmap doesn't seem to apply to this situation.

The initial data structure is like this:

{
chain: 'eth',
symbol: 'eth'
slip44: 60,
etc...
tokens: [ { symbol: fox, etc...} ]
}

The idea is to flatten it so the tokens are no longer nested under eth but are a flat list of all the assets (base assets and tokens) instead of having the tokens nested under the base asset.